larq/callbacks.py

```diff
@@ -40,6 +43,15 @@ def __init__(
         self.hyperparameter = hyperparameter
         self.verbose = verbose
 
+        # TODO: may want to make this a boolean instead for efficient comparison,
+        # need to test impact on speed.
```

Comment:
Since this just gets called once in the `__init__()`, I don't think we need to worry about speed.

Reply:
It's not the init I'm worried about; it's the string comparisons that potentially happen each batch. Just added the comment here because this is the place where we'd need to create that bool. I'll test whether this actually makes a difference though :+1:

Reply:
I have run an experiment using a per-step schedule, and the time per epoch is actually faster than it was before (while the setup otherwise remains identical), so I think it's safe to assume that there is no significant slowdown (of course I also strongly doubt that this actually caused a speedup, whatever effect this has is just outweighed by inherent variability in GPU time)